docs/docs/adrs/adr-015-partial-set-security.md

```diff
@@ -0,0 +1,308 @@
+a hook for [`AfterProposalVote`](https://github.com/cosmos/cosmos-sdk/blob/v0.47.7/x/gov/keeper/vote.go#L35) and keep track
+of all the votes cast by a validator. If a validator votes more than once, we only consider the latest vote.
+Finally, we only consider a validator has opted in if it casts a 100% `Yes` vote in case of a [weighted vote](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-037-gov-split-vote.md).
+
+
+### How do validators opt out?
+Validators that have opted in on a chain can opt out as well by sending the following message:
+
+```
+message MsgOptOut {
+    // the chain id of the consumer chain to opt out from
+    string chain_id = 1;
+    // the provider address of the validator on Cosmos Hub
+    string provider_addr = 2;
+}
+```
+Validators can only opt out after a consumer chain has started and hence the above message returns an error if the chain with `chain_id` is not running.
+
+#### State & Query
+We also update the state of the opted-in validators when a validator has opted out by removing it from there.
+Nevertheless, we have to keep track of all the validators that have opted in during the last `X` (to be defined) blocks to be able to jail validators for downtime.
+
+### When does a consumer chain start?
+A Top N consumer chain starts at the specified date (`spawn_time`) if the [`ConsumerAdditionProposal`](https://github.com/cosmos/interchain-security/blob/v3.3.0/proto/interchain_security/ccv/provider/v1/provider.proto#L27) has passed. An Opt In consumer chain starts if at least one validator has opted in. We check this in [BeginBlockInit](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/provider/keeper/proposal.go#L372):
+```
+func (k Keeper) BeginBlockInit(ctx sdk.Context) {
+    propsToExecute := k.GetConsumerAdditionPropsToExecute(ctx)
+
+    for _, prop := range propsToExecute {
+        chainId := prop.ChainId
+        if !k.IsTopN(ctx, chainId) && len(k.GetOptedInValidators(ctx, chainId)) == 0 {
+            // drop the proposal
+            ctx.Logger().Info("could not start chain because noone has opted in")
+            continue
+        }   
+        ...
+```
+
+### How do we send the partial validator sets to the consumer chains?
+To send the validator set to a consumer chain we first have to generate the validator updates with a function similar to this one:
+
+```
+func (k Keeper) GetPartialSetValidators(ctx types.Context, chainId string, valUpdates []abci.ValidatorUpdate) (newUpdates []abci.ValidatorUpdate)
+
+    powerSum := sdk.ZeroDec()
+    totalPower := k.ComputeTotalPower(ctx)
+    topNThreshold := k.GetTopN(ctx, chainId) / 100
+    for _, val := range valUpdates {
+        powerSum = powerSum.Add(sdk.NewDecFromInt(sdk.NewInt(val.Power)))
+        // if powerSum / totalPower > topNThreshold
+        if k.IsTopN(ctx, chainId) && powerSum.Quo(totalPower).GT(topNThreshold) {
```

Comment:
As discussed, we don't need to do anything with top n in here, since we do everything just by automatically opting validators in as discussed above

Reply:
Resolving this since we removed the partial-set computation from the ADR for now.